packages/ckeditor5-utils/tests/insertbypriority.js

```diff
@@ -0,0 +1,106 @@
+/**
+ * @license Copyright (c) 2003-2022, CKSource Holding sp. z o.o. All rights reserved.
+ * For licensing, see LICENSE.md or https://ckeditor.com/legal/ckeditor-oss-license
+ */
+
+import insertToPriorityArray from '../src/inserttopriorityarray';
+
+describe( 'insertToPriorityArray()', () => {
+	let objectsWithPriority;
+
+	beforeEach( () => {
+		objectsWithPriority = [];
+	} );
+
+	it( 'should insert only object to array', () => {
+		const objectA = { priority: 'normal' };
+
+		const expectedOutput = [ objectA ];
+
+		insertToPriorityArray( objectsWithPriority, objectA );
+
+		expect( objectsWithPriority ).to.deep.equal( expectedOutput );
+	} );
+
+	it( 'should place object with highest priority at the first index of an array', () => {
+		const objectA = { priority: 'high' };
+		const objectB = { priority: 'low' };
+
+		const expectedOutput = [ objectA, objectB ];
+
+		insertToPriorityArray( objectsWithPriority, objectA );
+		insertToPriorityArray( objectsWithPriority, objectB );
+
+		expect( objectsWithPriority ).to.deep.equal( expectedOutput );
+	} );
+
+	it( 'should place object with highest priority at the first index of an array even if inserted later', () => {
+		const objectA = { priority: 'high' };
+		const objectB = { priority: 'low' };
+
+		const expectedOutput = [ objectA, objectB ];
+
+		insertToPriorityArray( objectsWithPriority, objectA );
+		insertToPriorityArray( objectsWithPriority, objectB );
+
+		expect( objectsWithPriority ).to.deep.equal( expectedOutput );
+	} );
+
+	it( 'should correctly insert items by priority', () => {
+		const objectA = { priority: 'high' };
+		const objectB = { priority: 'lowest' };
+		const objectC = { priority: 'highest' };
+		const objectD = { priority: 'normal' };
+		const objectE = { priority: 'low' };
+
+		const expectedOutput = [ objectC, objectA, objectD, objectE, objectB ];
+
+		insertToPriorityArray( objectsWithPriority, objectA );
+		insertToPriorityArray( objectsWithPriority, objectB );
+		insertToPriorityArray( objectsWithPriority, objectC );
+		insertToPriorityArray( objectsWithPriority, objectD );
+		insertToPriorityArray( objectsWithPriority, objectE );
+
+		expect( objectsWithPriority ).to.deep.equal( expectedOutput );
+	} );
```

Comment:
I would suggest to duplicate this test with multiple items at the same priority but added in the mixed order.